File: 4.ROS/Tasks/Task_1/nomeer_ws/src/sub_pkg/scripts/sub.py
#!/usr/bin/python3

import rospy
from std_msgs.msg import String

def Lisener():
    rospy.init_node("lisener")
    sub = rospy.Subscriber("msg_topic", String, callback)
    rospy.spin()


def callback(msg):
    rospy.loginfo("%s", msg.data)
    

if __name__ == "__main__":
    Lisener()

File: 4.ROS/practice/nomeer1_ws/src/sub_pkg/src/sub.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

// defince callback function
void counterCallback(const std_msgs::Int32::ConstPtr& msg)
{
    ROS_INFO("%d", msg->data);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "tobic_subscriber"); // create node
    ros::NodeHandle nh;  // NodeHandle make ros component (node, topic) interact with ROS system

    // create subscriber object
    ros::Subscriber sub;
    sub = nh.subscribe("counter", 1000, counterCallback);
    ros::spin(); // create a loop that will keep the program in execution


    return 0;
}